device_backends/LogicalNameMapping/src/LNMBitRangeAccessPlugin.cc

```diff
@@ -0,0 +1,232 @@
+      }
+
+      this->_versionNumber = {};
+      this->_dataValidity = _target->dataValidity();
+    }
+
+    /********************************************************************************************************************/
+
+    void doPreWrite(TransferType type, VersionNumber /*versionNumber*/) override {
+      _lock.lock();
+
+      if(!_target->isWriteable()) {
+        throw ChimeraTK::logic_error("This register with BitRange plugin is not writeable.");
+      }
+
+      uint64_t value{};
+      if constexpr(std::is_same<UserType, ChimeraTK::Void>::value) {
+        // This should never be reached, the decorate function should make sure that we only have integral types here
+        assert(false);
+      }
+      else if constexpr(std::is_same<UserType, std::string>::value) {
+        // This should never be reached, the decorate function should make sure that we only have integral types here
+        assert(false);
+      }
+      else if constexpr(std::is_same<UserType, ChimeraTK::Boolean>::value) {
+        assert(false);
+      }
+      else {
+        memcpy(&value, buffer_2D[0].data(), sizeof(UserType));
+        value <<= _shift;
+      }
+
+      // When in a transfer group, only the first accessor to write to the _target can call read() in its preWrite()
+      // Otherwise it will overwrite the
+      if(_target->isReadable() && (!TransferElement::_isInTransferGroup || _lock.useCount() == 1)) {
+        _target->read();
+      }
+
+      _target->accessData(0) &= ~_mask;
+      _target->accessData(0) |= value;
+
+      _temporaryVersion = {};
+      _target->setDataValidity(this->_dataValidity);
+      _target->preWrite(type, _temporaryVersion);
+    }
+
+    /********************************************************************************************************************/
+
+    void doPostWrite(TransferType type, VersionNumber /*versionNumber*/) override {
+      auto unlock = cppext::finally([this] { this->_lock.unlock(); });
+      _target->postWrite(type, _temporaryVersion);
+    }
+
+    uint64_t _shift;
+    uint64_t _numberOfBits;
+    uint64_t _mask;
+
+    ReferenceCountedUniqueLock _lock;
+    VersionNumber _temporaryVersion;
+
+    using ChimeraTK::NDRegisterAccessorDecorator<UserType, TargetType>::_target;
+  };
+
+  /********************************************************************************************************************/
+
+  BitRangeAccessPlugin::BitRangeAccessPlugin(
+      const LNMBackendRegisterInfo& info, const std::map<std::string, std::string>& parameters)
+  : AccessorPlugin<BitRangeAccessPlugin>(info) {
+    _needSharedTarget = true;
+
+    try {
+      const auto& shift = parameters.at("shift");
+      auto [_, ec]{std::from_chars(shift.data(), shift.data() + shift.size(), _shift)};
+      if(ec != std::errc()) {
+        throw ChimeraTK::logic_error(
+            R"(LogicalNameMappingBackend BitRangeAccessPlugin: Unparseable parameter "shift".)");
```

Comment:
Please always mention the register name in exceptions. Also it would be helpful to include the erroneous string in the exception.

Reply:
Fixed